.github/workflows/upgrade-provider.yml

```diff
@@ -43,8 +47,9 @@ jobs:
       # uses: pulumi/pulumi-upgrade-provider-action@v0.0.10
       uses: ./.github/actions/upgrade-provider
       with:
-        kind: all
+        kind: provider
         email: noreply@github.com
         username: GitHub
         pr-assign: ${{ env.PR_ASSIGN }}
         pr-title-prefix: "feat: "
+        target-java-version: ${{ steps.pulumi-java.outputs.version }}
```

Comment:
Pursuant to comments on #130 I've also attempted to limit the scope of the `upgrade-provider` workflow to only upgrade `terraform-provider-equinix` and nothing else.  This seems to require 2 changes to the upgrade command / action:

1. Setting `kind` to `provider` to run an upstream provider upgrade
2. Setting `target-java-version` to the currently-used version of pulumi-java because the upgrade-provider command defaults to upgrading Java and the hidden flag for turning that off isn't exposed by the action

Command-line options are documented here: https://github.com/pulumi/upgrade-provider/tree/b625b8f79e1e57d625bbc31aadaefc380a917c16?tab=readme-ov-file#from-the-command-line